cpp/src/components/cuml_allocator.hpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
```

Comment:
This is just a comment, and I guess this is more of our technical debt, but bringing cuml allocator for just one algorithm largely breaks consistency of our codebase. Should be resolved once RAFT gets in place.

Reply:
I agree.  I needed to update the CUML code that had already been imported to do this to a newer version.  The best option I could find was to bring over the minimal set of functions to make it work.

I'm inclined (we can discuss separately) as we look at RAFT to seriously ponder how we want to handle this.

Reply:
Yes this is part of what we should discuss in RAFT and related to #727